### PR TITLE
LESQ-1005: Can't find your school form

### PR DESCRIPTION
**Pull request by k-huggs**
## Description

Music year: 1973

- Adds `ManualEntryComponent`
- Adds `SchoolDetailsInputComponent`
- Amends schema definition for school selection form
- Adds buttons to switch between forms
- Adds error states / components for forms
- onSubmit registers data to be connected to hubspot form

## Issue(s)

Do we want the values in the form to persist or refresh when changing page?

Acceptance criteria
--
- [ ] Add button to the school picker “Enter manually” that changes input options to “School name (required)” and “School address (required)” free text input
- [ ] Button “Select from dropdown” to take you back to school picker
- [ ] Errors if text inputs are empty

Fixes #LESQ-1005

## How to test

1. Go to https://deploy-preview-2709--oak-web-application.netlify.thenational.academy/onboarding
2. Click on yes, when posed with the do you work in a school question
3. You should see the following page
Screenshot 2024-08-16 at 13 49 51
4. Enter a school to use the school picker, the button should become enabled when school picker has a value and continue
4b. Select enter manually button
5. School name and school address input boxes should render, fill both of them for continue button to be available
Screenshot 2024-08-16 at 13 51 13
6. If you select from dropdown, returns to SchoolPicker component

## Screenshots

https://github.com/user-attachments/assets/7d8c9fda-5eeb-4c6c-8292-4562b4073060

## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner
- [ ] Does this PR update a package with a breaking change

### CODE REVIEW

**Comment by k-huggs** [comment]
Pushing up refactored functions without undefined values now  😬

File: src/components/TeacherComponents/ManualEntrySchoolDetails/ManualEntrySchoolDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  setRenderManualSchoolInput,
+  control,
+  onManualSchoolInputChange,
+
+  hasErrors,
+  reset,
+}) => {
+  return (
+    <>
+      {"manualSchoolName" in hasErrors && hasErrors.manualSchoolName && (
```
Comment (njblore) [inline src/components/TeacherComponents/ManualEntrySchoolDetails/ManualEntrySchoolDetails.tsx:35]: are these checking the same thing?

File: src/components/TeacherComponents/OnboardingForm/OnboardingForm.schema.ts [hunk 2]
```diff
@@ -18,17 +18,31 @@ export type RoleSelectFormProps = RoleSelectFormValues & {
...
+      .string({
+        errorMap: () => ({
+          message: "Select school",
+        }),
+      })
+      .min(1, "Select school"),
+    schoolName: z.string().optional(),
+    ...baseSchema.shape,
+  })
+  .or(
```
Comment (njblore) [inline src/components/TeacherComponents/OnboardingForm/OnboardingForm.schema.ts:33]: ooh nice I didn't know you could use `or` in a schema like this 🤌 is it possible to share the baseSchema between them so we don't need to duplicate it?

File: src/components/TeacherComponents/SchoolDetailsInputBox/SchoolDetailsInputBox.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  background,
+  labelText,
+  onBlur,
+  onChange,
+  hasError,
+}) => {
+  return (
+    <OakFlex $width={"100%"} $position={"relative"} $display={"inline-block"}>
+      <BoxBorders gapPosition="rightTop" />
+      <OakFlex $position={"absolute"}>
```
Comment (njblore) [inline src/components/TeacherComponents/SchoolDetailsInputBox/SchoolDetailsInputBox.tsx:31]: is this not the same as the `Input` component?

File: src/components/TeacherComponents/ManualEntrySchoolDetails/ManualEntrySchoolDetails.tsx (hunk 1 above)
Comment (k-huggs) [reply src/components/TeacherComponents/ManualEntrySchoolDetails/ManualEntrySchoolDetails.tsx]: removed second check ✅

File: src/components/TeacherComponents/OnboardingForm/OnboardingForm.schema.ts (hunk 2 above)
Comment (k-huggs) [reply src/components/TeacherComponents/OnboardingForm/OnboardingForm.schema.ts:33]: I think they're technically two different objects so needs to be twice

File: src/components/TeacherComponents/SchoolDetailsInputBox/SchoolDetailsInputBox.tsx (hunk 3 above)
Comment (k-huggs) [reply src/components/TeacherComponents/SchoolDetailsInputBox/SchoolDetailsInputBox.tsx:31]: Changed to Input component ✅

File: src/components/TeacherViews/Onboarding/SchoolSelection/SchoolSelection.view.tsx [hunk 4]
```diff
@@ -22,6 +31,32 @@ export const SchoolSelectionView = () => {
...
+      if (!isSchoolName) {
+        setValue("schoolAddress", value, {
+          shouldValidate: true,
+        });
+      }
+    },
+    [setValue],
+  );
+
+  const onManualSchoolInputChange = (isSchoolName: boolean, value: string) => {
```
Comment (njblore) [inline src/components/TeacherViews/Onboarding/SchoolSelection/SchoolSelection.view.tsx:51]: can this just be `setSchoolDetailsInManualForm(isSchoolName, value)` ?

**Review by njblore — APPROVED** [review]

**Review by mikeritson-oak — APPROVED** [review]
Work in this PR checked and passed. Form is present, functional and directs users between the school finder and manual input form, with or without error messages. Some minor styling work to be picked up with the main styling ticket (comment added).

**Pull request merged**